lib/pure/times.nim

```diff
@@ -744,6 +744,336 @@ proc format*(info: TimeInfo, f: string): string =
+
+proc parseToken(info: var TimeInfo; token, value: string; j: var int) =
+    ## Helper of the parse proc to parse individual tokens.
+    case token
+    of "d":
+        if value[j+1] in Digits:
+            # two digit format
+            info.monthday = value[j..j+1].parseInt()
+            j += 2
+        else:
+            info.monthday = parseInt($value[j])
+            j += 1
+    of "dd":
+        # two digit format
+        info.monthday = value[j..j+1].parseInt()
+        j += 2
+    of "ddd":
+        case value[j..j+2].toLower():
+        of "sun":
+            info.weekday = dSun
+        of "mon":
+            info.weekday = dMon
+        of "tue":
+            info.weekday = dTue
+        of "wed":
+            info.weekday = dWed
+        of "thu":
+            info.weekday = dThu
+        of "fri":
+            info.weekday = dFri
+        of "sat":
+            info.weekday = dSat
+        else:
+            raise newException(ValueError, "invalid day of week ")
+        j += 3
+    of "dddd":
+        if value.match(peg"^i'sunday'.*"):
+            info.weekday = dSun
+            j += 6
+        elif value.match(peg"^i'monday'.*"):
+            info.weekday = dMon
+            j += 6
+        elif value.match(peg"^i'tuesday'.*"):
+            info.weekday = dTue
+            j += 7
+        elif value.match(peg"^i'wednesday'.*"):
+            info.weekday = dWed
+            j += 9
+        elif value.match(peg"^i'thursday'.*"):
+            info.weekday = dThu
+            j += 8
+        elif value.match(peg"^i'friday'.*"):
+            info.weekday = dFri
+            j += 6
+        elif value.match(peg"^i'saturday'.*"):
+            info.weekday = dSat
+            j += 8
+        else:
+            raise newException(ValueError, "invalid day of week ")    
+    of "h", "H":
+        if value[j+1] in Digits:
+            # two digit format
+            info.hour = value[j..j+1].parseInt()
+            j += 2
+        else:
+            info.hour = parseInt($value[j])
+            j += 1
+    of "hh", "HH":
+        # two digit format
+        info.hour = value[j..j+1].parseInt()
+        j += 2
+    of "m":
+        if value[j+1] in Digits:
+            # two digit format
+            info.minute = value[j..j+1].parseInt()
+            j += 2
+        else:
+            info.minute = parseInt($value[j])
+            j += 1
+    of "mm":
+        # two digit format
+        info.minute = value[j..j+1].parseInt()
+        j += 2
+    of "M":
+        var month: int
+        if value[j+1] in Digits:
+            # two digit format
+            month = value[j..j+1].parseInt()
+            j += 2
+        else:
+            month = parseInt($value[j])
+            j += 1
+        info.month = getMonth(month)
+    of "MM":
+        var month = value[j..j+1].parseInt()
+        j += 2
+        info.month = getMonth(month)
+    of "MMM":
+        case value[j..j+2].toLower():
+        of "jan":
```

Comment:
I couldn't get this to work. When I tried it was expecting a full month name versus the 3 letter month. In other words parseEnum[Month]("december") works, but not parseEnum[Month]("mDec"). Am I missing something?